e2e-tests/trade-finance-manager/cypress/integration/journeys/case/underwriting/pricing-and-risk-risk-managers.spec.js

```diff
@@ -0,0 +1,160 @@
+      pages.underwritingPricingAndRiskEditPage.creditRatingTextInputOther().should('be.visible');
+      pages.underwritingPricingAndRiskEditPage.creditRatingTextInputOtherValidationError().should('be.visible');
+    });
+
+    it('submitting a rating displays the rating in table on `pricing and risk` page and does not render `add credit rating` link', () => {
+      pages.underwritingPricingAndRiskPage.addRatingLink().click();
+
+      // select option, submit
+      pages.underwritingPricingAndRiskEditPage.creditRatingRadioInputGood().click();
+      pages.underwritingPricingAndRiskEditPage.submitButton().click();
+
+      // assert elements/value in `pricing and risk` page
+      cy.url().should('eq', relative(`/case/${dealId}/underwriting/pricing-and-risk`));
+
+      pages.underwritingPricingAndRiskPage.addRatingLink().should('not.exist');
+
+      pages.underwritingPricingAndRiskPage.exporterTableChangeCreditRatingLink().should('be.visible');
+
+      pages.underwritingPricingAndRiskPage.exporterTableRatingValue().invoke('text').then((text) => {
+        expect(text.trim()).to.equal('Good (BB-)');
+      });
+    });
+
+    it('after submitting a rating, editing the rating has default value and new rating displays in `pricing and risk` page', () => {
+      // check value previously submitted
+      pages.underwritingPricingAndRiskPage.exporterTableRatingValue().invoke('text').then((text) => {
+        expect(text.trim()).to.equal('Good (BB-)');
+      });
+
+      pages.underwritingPricingAndRiskPage.exporterTableChangeCreditRatingLink().click();
+
+      // previously submitted value should be auto selected
+      pages.underwritingPricingAndRiskEditPage.creditRatingRadioInputGood().should('be.checked');
+
+      // submit different value
+      pages.underwritingPricingAndRiskEditPage.creditRatingRadioInputAcceptable().click();
+      pages.underwritingPricingAndRiskEditPage.submitButton().click();
+
+      cy.url().should('eq', relative(`/case/${dealId}/underwriting/pricing-and-risk`));
+
+      // check new value displays in `pricing and risk` page
+      pages.underwritingPricingAndRiskPage.exporterTableRatingValue().invoke('text').then((text) => {
+        expect(text.trim()).to.equal('Acceptable (B+)');
+      });
+    });
+  });
+
+  describe('Risk Managers can edit once logged in', () => {
+    beforeEach(() => {
+      cy.login(riskManagerUser);
+
+      cy.visit(`/case/${dealId}/underwriting/pricing-and-risk`);
+      cy.url().should('eq', relative(`/case/${dealId}/underwriting/pricing-and-risk`));
+    });
+
+    it('edit button is visible as compared to add', () => {
+      pages.underwritingPricingAndRiskPage.addRatingLink().should('not.exist');
+      pages.underwritingPricingAndRiskPage.exporterTableChangeCreditRatingLink().should('exist');
+      pages.underwritingPricingAndRiskPage.exporterTableChangeProbabilityOfDefaultLink().should('exist');
+    });
+  });
+});
```

Comment:
I think we can simplify this file a bit - most of this is already tested in the other pricing-and-risk E2E test.

Should we have just the last 3 tests in here, so it's only testing the submission and after submission?

I think we should remove the tests before this one:
```submitting a rating displays the rating in table on `pricing and risk` page and does not render `add credit rating` link```

Just thinking of E2E test time, the more tests we have the longer it takes

What do you think?

Reply:
think good idea, have done it